docs/cvl/rules.md

```diff
@@ -219,6 +219,128 @@ example and run the rule on the example.  Instead, the Prover translates the
+      uint256 p2 = previewDeposit(e3, assets);
+
+      _AToken.approve(e4, currentContract, 0);
+      require _AToken.allowance(currentContract, user) < assets;
+      uint256 p3 = previewDeposit(e5, assets);
+
+      assert p1 == p2 && p2 == p3, "preview ignores allowance";
+  }
+  ```
+
+- Joining and splitting near-additivity. When conversions round, splitting/merging accounts yield off-by-one envelopes. Use mathint to avoid silent overflow:
+
+  ```cvl
+  /// Convert sum of assets is within [parts, parts+1]
+  rule convertSumOfAssetsPreserved(uint256 a1, uint256 a2) {
+      env e;
+      uint256 s1 = convertToShares(e, a1);
+      uint256 s2 = convertToShares(e, a2);
+      uint256 as = require_uint256(a1 + a2);
+      mathint js = convertToShares(e, as);
+      assert js >= s1 + s2;
+      assert js <  s1 + s2 + 2;
+  }
+  ```
+
+- Deposit envelopes by index (Aave-style index RAY). Bound deposited aTokens relative to requested `assets` and index. For `index > RAY` the bound is `+1 aToken`; for `index == RAY` it tightens to `+0.5 aToken`:
+
+  ```cvl
+  rule depositUpperBound(env e){
+      uint256 assets; address receiver;
+      uint256 before = _AToken.balanceOf(currentContract);
+      uint256 idx = _SymbolicLendingPool.getReserveNormalizedIncome(asset());
+      require e.msg.sender != currentContract;
+      uint256 shares = deposit(e, assets, receiver);
+      uint256 after  = _AToken.balanceOf(currentContract);
+      assert (idx > RAY()  => after - before <= assets + idx / RAY());
+      assert (idx == RAY() => after - before <= assets + idx / (2 * RAY()));
+  }
+  ```
+
+- Non-zero mint condition. Ensure users receive at least one share when depositing at least one unit in index terms:
+
+  ```cvl
+  rule depositMintsAtLeastOne(env e){
+      uint256 assets; address receiver;
+      uint256 idx = _SymbolicLendingPool.getReserveNormalizedIncome(asset());
+      require e.msg.sender != currentContract;
+      uint256 shares = deposit(e, assets, receiver);
+      assert assets * RAY() >= to_mathint(idx) => shares != 0;
+  }
+  ```
+
+- Mint envelopes. When minting shares directly, the receiver’s balance increases by the requested amount, up to an extra unit due to rounding:
+
+  ```cvl
+  rule mintBounds(env e){
+      uint256 shares; address receiver;
+      require e.msg.sender != currentContract;
+      uint256 idx  = _SymbolicLendingPool.getReserveNormalizedIncome(asset());
+      uint256 pre  = balanceOf(e, receiver);
+      require pre + shares <= max_uint256; // avoid overflow in spec
+      uint256 assets = mint(e, shares, receiver);
+      uint256 post = balanceOf(e, receiver);
+      assert (idx >= RAY());
+      assert to_mathint(post) >= pre + shares;
+      assert to_mathint(post) <= pre + shares + 1;
+  }
+  ```
+
+- Duplicate reward claims do not amplify payout. Whether the contract has sufficient funds or not, listing the same reward twice must not increase net rewards to a user beyond the computed claimable amount:
```

Comment:
this is a legitimate rule but it's in the wrong section. Again, I think it should be in an examples repository for reward-distributing code